plugin/trace/trace.go

```diff
@@ -54,7 +54,12 @@ func (t *trace) OnStartup() error {
 		case "zipkin":
 			err = t.setupZipkin()
 		case "datadog":
-			tracer := opentracer.New(tracer.WithAgentAddr(t.Endpoint), tracer.WithServiceName(t.serviceName), tracer.WithDebugMode(log.D.Value()))
+			tracer := opentracer.New(
+				tracer.WithAgentAddr(t.Endpoint),
+				tracer.WithServiceName(t.serviceName),
+				tracer.WithDebugMode(log.D.Value()),
+				tracer.WithSampler(tracer.NewRateSampler(1/float64(t.every))),
```

Comment:
Much cleaner 👍 .

Reply:
I don't want to use a complex sampler, because the default sampler for zipkin is also very simple.

If we want to add additional functionality such as a more complex sampler, it should be in another PR altogether.
With the explicit necessity for it.
It's a feature by itself.

Reply:
I would not use this sampler. It will stop spans from reaching the agent and as a result APM stats will be skewed. This will cause an incomplete experience. We generally don't use this sampler (and should likely deprecate it).

Instead, use the [`WithSamplingRules`](https://godoc.org/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer#WithSamplingRules) option along with the [`RateRule`](https://godoc.org/gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer#RateRule)

Reply:
I'm fine with it as long as behavior is the same.

Reply:
+1 with switching to `WithSamplingRules`.